modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueArea.scala

```diff
@@ -126,32 +130,49 @@ object QueueTableBody {
 
   private val PhoneCut = 400
   private val LargePhoneCut = 570
-  private val IconColumnWidth = 12
+  private val IconColumnWidth = 20
   private val ObsIdColumnWidth = 140
   private val StateColumnWidth = 80
+  private val InstrumentColumnWidth = 80
+  private val NameColumnWidth = 140
   private val TargetNameColumnWidth = 140
+  private val ColumnPadding = 10 + 1 // Taken from SeqexecStyles.queueText padding and 1 for border
 
   val statusHeaderRenderer: HeaderRenderer[js.Object] = (_, _, _, _, _, _) =>
     <.div(
       ^.title := "Control",
       ^.width := IconColumnWidth.px
     )
 
+  val QueueColumnStyle: String = SeqexecStyles.queueTextColumn.htmlClass
+
   private def columns(p: Props, s: Size): List[Table.ColumnArg] = {
     val isLogged = p.sequences().isLogged
-    val targetColumn = Column(Column.props(TargetNameColumnWidth, "target", flexShrink = 1, flexGrow = 3, label = "Target", cellRenderer = targetRenderer(p), className = SeqexecStyles.queueTextColumn.htmlClass))
-    val nameColumn = Column(Column.props(TargetNameColumnWidth, "obsName", flexShrink = 1, flexGrow = 3, label = "Obs. Name", cellRenderer = obsNameRenderer(p), className = SeqexecStyles.queueTextColumn.htmlClass))
-    val (loggedInWidth, loggedInColumns) = s.width match {
-      case w if w < PhoneCut      => (0, Nil)
-      case w if w < LargePhoneCut => (TargetNameColumnWidth, List(targetColumn))
-      case _                      => (2 * TargetNameColumnWidth, List(targetColumn, nameColumn))
+
+    // Calculate the column's width
+    val (obsColumnWidth, statusColumnWidth, instrumentColumnWidth, nameColumnWidth, targetNameColumnWidth) = p.sequences().sequences.map {
+      case SequenceInQueue(id, st, i, _, n, t, r) =>
+        (tableTextWidth(id), tableTextWidth(statusText(st, r)), tableTextWidth(i.shows), tableTextWidth(n), tableTextWidth(t.getOrElse("")))
+    }.foldLeft((ObsIdColumnWidth, StateColumnWidth, InstrumentColumnWidth, NameColumnWidth, TargetNameColumnWidth)) {
+      case ((o, s, i, n, t), (co, cs, ci, cn, ct)) =>
```

Comment:
I admit that I find this part (with so many variables and the `foldLeft` a little bit confusing, but the end result looks very good, so I'll give this a 👍 .